lib/exchange/adapters/rabbitmq/consumer.ex

```diff
@@ -0,0 +1,157 @@
+defmodule Exchange.Adapters.RabbitBus.Consumer do
+  @moduledoc """
+  This module is a server and consumer of events.
+  This server handles the addition and removal of listener processes.
+  It consumes messages from a RabbitMQ queue, decodes the message and then the message is broadcasted to every listener.
+  This module manages the necessary resources to use the RabbitMQ
+  """
+  use GenServer
+  require Logger
+  alias AMQP.{Basic, Connection}
+
+  @host "amqp://localhost"
+  @reconnect_interval 10_000
+
+  def start_link(_) do
+    GenServer.start_link(__MODULE__, %{chan: nil, subs: %{}}, name: :rabbitmq_consumer)
+  end
+
+  def init(state) do
+    send(self(), :connect)
+    {:ok, state}
+  end
+
+  @doc """
+  Removes a subscriber of event
+  """
+  def handle_call({:remove_listener, event, subscriber}, _, %{chan: _, subs: subs} = state) do
+    event_subs = Map.get(subs, event, [])
+    event_subs = List.delete(event_subs, subscriber)
+    subs = Map.put(subs, event, event_subs)
+    state = Map.put(state, :subs, subs)
```

Comment:
done in one line with `put_in/2` or `put_in/3` https://hexdocs.pm/elixir/Kernel.html#put_in/2